network/dag/publisher.go

```diff
@@ -176,13 +179,22 @@ func (s *replayingDAGPublisher) publishTransaction(ctx context.Context, transact
 		return false
 	}
 
-	for _, payloadType := range []string{transaction.PayloadType(), AnyPayloadType} {
-		receiver := s.subscribers[payloadType]
-		if receiver == nil {
-			continue
-		}
-		if err := receiver(transaction, payload); err != nil {
-			log.Logger().Errorf("Transaction subscriber returned an error (ref=%s,type=%s): %v", transaction.Ref(), transaction.PayloadType(), err)
+	// TODO: Now calls TransactionAddedEvent and TransactionPayloadAddedEvent after checken whether the payload is present.
```

Comment:
Also, the TransactionAddedEvents may be blocked by a missing payload of a did.Document transaction